fuzzing/src/visitors/facevisitor-colrv1.cpp

```diff
@@ -0,0 +1,245 @@
+// facevisitor-colrv1.cpp
+//
+//   Finds and traverses COLRv1 glyphs using FreeType's
+//   COLRv1 API.
+//
+// Copyright 2021 by
+// Dominik Röttsches.
+//
+// This file is part of the FreeType project, and may only be used,
+// modified, and distributed under the terms of the FreeType project
+// license, LICENSE.TXT.  By continuing to use, modify, or distribute
+// this file you indicate that you have read the license and
+// understand and accept it fully.
+
+
+#include "visitors/facevisitor-colrv1.h"
+
+#include <cassert>
+
+#include "utils/logging.h"
+
+#include <ft2build.h>
+#include FT_FREETYPE_H
+
+namespace {
+
+bool colrv1_start_glyph( const FT_Face& ft_face,
+                         uint16_t glyph_id,
+                         FT_Color_Root_Transform root_transform );
+
+void iterate_color_stops ( FT_Face face, FT_ColorStopIterator* color_stop_iterator ) {
```

Comment:
let's please make those private class members 😊

Reply:
May I ask what's the reasoning behind this suggestion? These are pure helper functions implementing the recursive traversal of the COLRv1 glyph graph, they're not accessing class members of `FaceVisitorColrV1` and are local to this `FaceVisitor`. FWIW the Google C++ style guide [suggests prefering internal linkage](https://google.github.io/styleguide/cppguide.html#Internal_Linkage)). Seems to me it'd make sense to keep them in this anonymous namespace.

Reply:
oh sorry missed this comment before I answered below; from my point of view converting free functions to static class members helps with "finding" their declarations e.g. in automated documentations or simply by browsing class definitions.

If I read the internal linkage paragraph correctly, it mainly suggests to not put declarations of module-private free functions in header files for public access (which I agree with completley). The paragraph below "Nonmember, Static Member, and Global Functions" leaves a bit more room about whether to use free functions or static members. In this case, "Static methods of a class should generally be closely related to instances of the class" seems to be fulfilled from my point of view.

In C++ projects I have worked on so far (commerical and private) I've always found that static members keep the code cleaner + easier to follow (one place to look for declarations, one place to look for definitions) but not too fussed here. Happy to leave it as is if you prefer that 😊

Reply:
> If I read the internal linkage paragraph correctly, it mainly suggests to not put declarations of module-private free functions in header files for public access (which I agree with completley). The paragraph below "Nonmember, Static Member, and Global Functions" leaves a bit more room about whether to use free functions or static members. In this case, "Static methods of a class should generally be closely related to instances of the class" seems to be fulfilled from my point of view.

I read the main clause 

> When definitions in a .cc file do not need to be referenced outside that file, give them internal linkage by placing them in an unnamed namespace or declaring them static. 

to really mainly focus on linkage and not about public access / interface questions. I recommends to give only internal linkage to anything that's not referenced/used outside the implementation file - which in our case applies to the COLRv1 traversal function implementations. 

That's also common practice in Blink for file-local helper functions etc. - so if you're okay with that I am happy to leave them there. I don't have too strong of an opinion on it either. I was mainly curious what's the background of the request.

Reply:
thanks for the extened conversation on this 😊 all the times I've built CEF recently, I haven't looked into the code yet ... do you happen to know the rough % of static helpers vs. class member functions in Blink? would be interesting to know whether building (esp. linking) would take even longer without strong focus on internal linking ... only downside I think to see is code clarity ("finding" relevant pieces directly in the code or in automated docs) ... might be wrong though + could just be a matter of getting used to it. an interesting thought regardless; will keep that in mind for future projects!

TL;DR let's leave it as is then 😊 only suggestion I had would be to move the `constexpr` definition to the top to roughly follow the pattern of "variables first, then functions" (suggeseted that in my gist)